gpu/people/tools/people_tracking.cpp

```diff
@@ -72,12 +73,27 @@ class PeopleTrackingApp
       interface->stop ();
     }
 
-    pcl::visualization::CloudViewer         viewer;
-    pcl::people::trees::MultiTreeLiveProc*  m_proc;
-    cv::Mat                                 m_lmap;
-    cv::Mat                                 m_cmap;
-    cv::Mat                                 cmap;
-    cv::Mat                                 m_bmap;
+    void load_tree(std::string treeFilenames[4], int numTrees)
+    {
+      std::ifstream fin0 (treeFilenames[0]);
+      assert(fin0.is_open());
+      m_proc = pcl::make_shared<pcl::people::trees::MultiTreeLiveProc>(fin0);
+
+      /// Load the other tree files
+      for (const auto& file : treeFilenames)
+      {
+        std::ifstream fin (file);
+        assert (fin.is_open());
+        m_proc->addTree(fin);
+      }
+    }
+
+    pcl::visualization::CloudViewer                           viewer;
+    boost::shared_ptr<pcl::people::trees::MultiTreeLiveProc>  m_proc;
```

Comment:
```suggestion
    std::unique_ptr<pcl::people::trees::MultiTreeLiveProc>  m_proc;
```

There's no reason to use a shared pointer here. The class is not being passed around. This is a tool and not part of the library interface so we can use std pointers directly.

Reply:
I realized I've just contradicted my previous review. I'm terribly sorry. I came with a fresh pair of eyes without looking at the prior comments and noticed some inconsistencies. Again sorry for the extra noise.

Reply:
No problem at all. Just to clarify, I should convert `m_proc` to `std::unique_ptr`, right?

Reply:
Exactly.